lnwire/musig2.go

```diff
@@ -7,21 +7,33 @@ import (
 	"github.com/lightningnetwork/lnd/tlv"
 )
 
-const (
-	// NonceRecordType is the TLV type used to encode a local musig2 nonce.
-	NonceRecordType tlv.Type = 4
-)
+// NonceRecordTypeT is the TLV type used to encode a local musig2 nonce.
+type NonceRecordTypeT = tlv.TlvType4
```

Comment:
not sure we should do this cause we wont always use type 4 each time a musig2 nonce is in a message. for example, gossip 1.75 stuff: https://github.com/lightningnetwork/lnd/pull/8044/commits/ced7497968849c0b6c61768ccd051525fcc5e492

Reply:
also feels like this is reverting back to the old way of defining the tlv type numbers instead of defining it in the message struct. To me, the exact type number only applies in the context of the message that the field is used in.

Reply:
yeah that is kinda what im advocating for. I know it seems silly now when all the types are type 4, but with gossip 1.75, we do actually use a different type numbers for a couple of nonces. 

but anyways, i can always just rename there when we actually need that. Just thought i'd mention in order to reduce refactor commits later on.

Reply:
>  but with gossip 1.75, we do actually use a different type numbers for a couple of nonces.

Yeah if each of the messages actually use distinct types (not the case anywhere in the spec _yet_) for the same record, then I think it makes sense.